### DISCUSSION TITLE
Issue retrieving media after upgrade from v8 to v10

### DISCUSSION THREAD

**Discussion by robbiemoore** [spatie/laravel-medialibrary#3088] [Q&A]
```
SQLSTATE[42883]: Undefined function: 7 ERROR:  operator does not exist: character varying = integer
  LINE 1: select * from "media" where "media"."model_id" in (2) and "m...
                                                         ^
  HINT:  No operator matches the given name and argument types. You might need to add explicit type casts. (SQL: select * from "media" where "media"."model_id" in (2) and "media"."model_type" = App\Models\ExampleModel)
```

When retrieving the media record in any manner. I think the issue stems from trying to morph to a combination of records that have string IDs and some with integer IDs. This was working perfectly before the upgrade, and this issue still is present after following the upgrade instructions. Any suggestions are gratefully received!

**Comment by systemsolutionweb** [comment]
> records that have string IDs and some with integer IDs

There is the problem
